shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSpherePreparedStatement.java

```diff
@@ -307,7 +307,9 @@ private ResultSet executeFederationQuery(final QueryContext queryContext) throws
     }
     
     private DriverExecutionPrepareEngine<JDBCExecutionUnit, Connection> createDriverExecutionPrepareEngine() {
-        int maxConnectionsSizePerQuery = metaDataContexts.getMetaData().getProps().<Integer>getValue(ConfigurationPropertyKey.MAX_CONNECTIONS_SIZE_PER_QUERY);
+        int maxConnectionsSizePerQuery = connection.isHoldTransaction()
```

Comment:
Did you consider the possible extra memory usage when maxConnectionsSizePerQuery = 1?

Reply:
> Did you consider the possible extra memory usage when maxConnectionsSizePerQuery = 1?

Correctness is the most fundamental requirement.

Reply:
The ShardingSphere local transaction manager does not guarantee transaction correctness when routing to multiple data nodes. If required, use XA transactions to guarantee transactional correctness across multiple data nodes.

Reply:
@I-Love-China 
In the ss design, local does not guarantee any features. If you need to ensure transactionality, I think you need to develop some functions yourself. For example, if multiple data sources are of the same instance, you can use only one connection, and it is forbidden to perform cross-shard queries in transactions (I don't think it is necessary to perform a large batch of queries in a transaction, because you can already determine what shards you want to update, which needs to be acceptable to your business). If your data is small, you can also consider loading all into memory.

Reply:
> The ShardingSphere local transaction manager does not guarantee transaction correctness when routing to multiple data nodes. If required, use XA transactions to guarantee transactional correctness across multiple data nodes.

But these data nodes are located at the same database instance.

Reply:
> @I-Love-China In the ss design, local does not guarantee any features. If you need to ensure transactionality, I think you need to develop some functions yourself. For example, if multiple data sources are of the same instance, you can use only one connection, and it is forbidden to perform cross-shard queries in transactions (I don't think it is necessary to perform a large batch of queries in a transaction, because you can already determine what shards you want to update, which needs to be acceptable to your business). If your data is small, you can also consider loading all into memory.

I am so confused. According to this, local transactions will be totally destroyed.